### PR TITLE
vim-patch:8.2.{3451,3458,3452,3685,3686}: some filetypes are not recognized correctly

### PR DESCRIPTION
**Pull request by clason**
vim-patch:8.2.3451: not all apache files are recognized

Problem:    Not all apache files are recognized.
Solution:   Adjust the filetype pattern. (Zdenek Dohnal, closes vim/vim#8882)
https://github.com/vim/vim/commit/770ddbe1595f6dab836304203d5ca2e0b069266f

vim-patch:8.2.3458: not all dictdconf files are recognized

Problem:    Not all dictdconf files are recognized.
Solution:   Adjust the pattern. (Doug Kearns)
https://github.com/vim/vim/commit/be01090efad242e30728275dea05420db6f96257

vim-patch:8.2.3452: MPD files are not recognized

Problem:    MPD files are not recognized.
Solution:   Recognize MPD files as XML. (Steven Penny, closes vim/vim#8893)
https://github.com/vim/vim/commit/690c524ce6629f9ff67728541ba211f831caf0ee

vim-patch:8.2.3685: Visual studio project files are not recognized

Problem:    Visual studio project files are not recognized.
Solution:   Use the xml file type. (Doug Kearns)
https://github.com/vim/vim/commit/c07f11e42fc2eac5e750bf05aa3030f9b02a22ca

vim-patch:8.2.3686: filetype detection often mixes up Forth and F#

Problem:    Filetype detection often mixes up Forth and F#.
Solution:   Add a function to inspect the file contents. (Doug Kearns)
https://github.com/vim/vim/commit/3d14c0f2b964195d08b34bb43f89ec5f99255194

### CODE REVIEW

File: src/nvim/testdir/test_filetype.vim [hunk 1]
```diff
@@ -553,7 +553,7 @@ let s:filename_checks = {
     \ 'xhtml': ['file.xhtml', 'file.xht'],
     \ 'xinetd': ['/etc/xinetd.conf', '/etc/xinetd.d/file', 'any/etc/xinetd.conf', 'any/etc/xinetd.d/file'],
     \ 'xmath': ['file.msc', 'file.msf'],
-    \ 'xml': ['/etc/blkid.tab', '/etc/blkid.tab.old', 'file.xmi', 'file.csproj', 'file.csproj.user', 'file.ui', 'file.tpm', '/etc/xdg/menus/file.menu', 'fglrxrc', 'file.xlf', 'file.xliff', 'file.xul', 'file.wsdl', 'file.wpl', 'any/etc/blkid.tab', 'any/etc/blkid.tab.old', 'any/etc/xdg/menus/file.menu'],
+    \ 'xml': ['/etc/blkid.tab', '/etc/blkid.tab.old', 'file.xmi', 'file.csproj', 'file.csproj.user', 'file.fsproj', 'file.fsproj.user', 'file.vbproj', 'file.vbproj.user', 'file.ui', 'file.tpm', '/etc/xdg/menus/file.menu', 'fglrxrc', 'file.xlf', 'file.xliff', 'file.xul', 'file.wsdl', 'file.wpl', 'any/etc/blkid.tab', 'any/etc/blkid.tab.old', 'any/etc/xdg/menus/file.menu', 'file.atom', 'file.rss', 'file.cdxml', 'file.psc1', 'file.mpd'],
```
Comment (seandewar) [inline src/nvim/testdir/test_filetype.vim:557]: Good fix. Looks like this was accidentally reverted a bit in #14424.

**Review by seandewar — APPROVED** [review]
LGTM.
Nice job spotting the duplicate GPROF help section. :+1:

(The PR title is missing 3686 btw)

File: src/nvim/testdir/test_filetype.vim (hunk 1 above)
Comment (clason) [reply src/nvim/testdir/test_filetype.vim:557]: Yep, that had me confused for a bit :)

**Comment by clason** [comment]
(`make` complained loudly enough ;))

**Pull request merged**